### PR TITLE
Temporary fix for flaky tests

### PR DESCRIPTION
**Pull request by richard-cox**
### Summary
- Temporary fix for flaky user avatar based tests
- Added a way to skip the build
- Added tags for setup and flaky tests

### Occurred changes and/or fixed issues
- Temporary fix for flaky user avatar based tests
  - It's not an ideal solution, but added 2 x 1 second `wait`s to fix flaky tests
    - is quicker than everyone re-trying e2e gate multiple times
    - brings down e2e gate total time to ~55 minutes 
  - The tests have run a total of 8 times (3 as full suite) and all have passed first time.

Also....
- Added an optional, off by default, way to skip the build step and grab the dashboard from CDN
  - `BUILD_DASHBOARD: false`
  - As the code isn't instrumented the code cov parts are skipped
  - This is only good for debugging e2e tests in GH
- Added setup and flaky to some tests.
  - Can be used in conjunction with `BUILD_DASHBOARD` for super quick test runs (15mins)
    - admin tests env `GREP_TAGS: '@setup'`
    - standard user tests env `GREP_TAGS: '@flaky'` 
- Fix failures in deployment spec when no tests run

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "e2e:pre-dev": "yarn docker:local:stop && yarn docker:local:start && NODE_ENV=dev TEST_INSTRUMENT=true yarn build",
     "e2e:dev": "START_SERVER_AND_TEST_INSECURE=1 server-test start:dev https-get://localhost:8005 cy:run:sorry",
     "e2e:pre-prod": "yarn docker:local:stop && mkdir dist && TEST_INSTRUMENT=true ./scripts/build-e2e && ./scripts/e2e-docker-start ",
-    "e2e:prod": "GREP_TAGS=$GREP_TAGS TEST_USERNAME=$TEST_USERNAME TEST_BASE_URL=https://127.0.0.1/dashboard yarn cy:run:sorry",
+    "e2e:prod": "BUILD_DASHBOARD=$BUILD_DASHBOARD GREP_TAGS=$GREP_TAGS TEST_USERNAME=$TEST_USERNAME TEST_BASE_URL=https://127.0.0.1/dashboard yarn cy:run:sorry",
```
Comment (richard-cox) [inline package.json:46]: Check what happens when this is undefined

**Review by nwmac — APPROVED** [review]
Approved - will come back to the use of waits, but for now as a fix, this will improve reliability.

**Comment by cnotv** [comment]
@richard-cox since I took the effort to check all the issues one by one and document them, can you please provide a link where we ensure this to have been fixed before close? Otherwise it's hard to track everything, thanks.
Also we may want to link the PR to an issue, since is the standard workflow we have?

**Comment by richard-cox** [comment]
All the issues that were closed had a comment referencing this PR. The gates for this PR have run 3 times, plus additional runs where the tests were limited to just the flaky ones in question.

I'll add the fixes notation to provide a clearer link

**Pull request merged**